platform/osx/os_osx.mm

```diff
@@ -602,11 +602,14 @@ static void _mouseDownEvent(NSEvent *event, int index, int mask, bool pressed) {
 	Ref<InputEventMouseButton> mb;
 	mb.instance();
 
-	get_key_modifier_state([event modifierFlags], mb);
-	mb->set_button_index(index);
-	mb->set_pressed(pressed);
-	mb->set_position(Vector2(mouse_x, mouse_y));
-	mb->set_global_position(Vector2(mouse_x, mouse_y));
+    const CGFloat backingScaleFactor = [[event window] backingScaleFactor];
+    const Vector2 pos = get_mouse_pos([event locationInWindow], backingScaleFactor);
```

Comment:
If there's no mouse move on touchscreen, you should set `mouse_x`, `mouse_y` and `prev_mouse_x`, `prev_mouse_y` to the new values as these are used in multiple other places.

```c++
prev_mouse_x = mouse_x;
prev_mouse_y = mouse_y;

mouse_x = pos.x;
mouse_y = pos.y;
```

Reply:
I thought ```get_mouse_pos``` already does this. 
https://github.com/godotengine/godot/blob/c3ad83142ed5efd4d72495b58d852cb38a85554e/platform/osx/os_osx.mm#L115
Ok no, I see it does set mouse_x and mouse_y but not prev_mouse. This is a little bit strange IMO that the function automatically sets the current mouse position but not the old one. But I could set prev mouse the same way as mouse moved does?
https://github.com/godotengine/godot/blob/c3ad83142ed5efd4d72495b58d852cb38a85554e/platform/osx/os_osx.mm#L648

Reply:
Yep, I missed it, and `prev_mouse_*` seems to be unused at all.

Reply:
oh, now I have already changed it. But I could also remove prev_mouse?

Reply:
You can remove `prev_mouse` variables completely, also check your indentation, Godot's code style uses tabs not spaces.

https://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html
https://travis-ci.org/godotengine/godot/jobs/502512365 - clang format errors

Reply:
Ok I'll remove it and also fix the indentation. I use tabs too but I think xcode automatically converts them or something...